beta/src/components/Search.tsx

```diff
@@ -128,8 +128,23 @@ export function Search({
   );
 
   const onClose = useCallback(
-    function onClose() {
-      setIsShowing(false);
+    function onClose(event: any) {
+      if (event.ctrlKey || event.metaKey) {
+      } else {
+        const targetElement = event.target;
+        const clickedOnLink =
+          targetElement.classList.contains('DocSearch-Hit') ||
```

Comment:
These hardcoded class checks aren't great and look fragile. What are you trying to solve here? Can you think of another approach?

Reply:
I'm trying to think of better approach. Here, first i added a parent div to the around the `DocSearchModal` to add an `onClick` listener, to check if the person did cmd+click or normal click so as to not close the search bar if it was cmd+click. Then, as @shanpriyan mentioned that the search bar is getting closed whenever clicked anywhere inside it. So, i started searching for the child classes so as to **close the sidebar only if that was a normal click on the search result and not cmd+click**.